packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -301,7 +301,7 @@ export function createHooksInternal<
             (...args: any[]) => {
               const [queryKey, ...rest] = args;
               return queryClient.invalidateQueries(
-                getArrayQueryKey(queryKey),
+                getArrayQueryKey(queryKey, 'any'),
```

Comment:
I think this is prefect and that we don't need to allow granular invalidation for now, I don't see why anyone would _not_ want to invalidate the same query across both methods of fetching it.